### DISCUSSION TITLE
想用juicefs作为hdfs的对象存储，是否操作错误？

### DISCUSSION THREAD

**Discussion by ssq001** [juicedata/juicefs#1662] [Q&A] [answered]
#########  juicefs format --storage hdfs --bucket 192.168.56.3:9000 --access-key hdfs redis://:123456@127.0.0.1:6379/2 pics1

2022/03/28 13:57:40.690709 juicefs[2808338] <INFO>: Meta address: redis://:****@127.0.0.1:6379/2 [interface.go:383]
2022/03/28 13:57:40.692853 juicefs[2808338] <WARNING>: parse info: ERR AUTH <password> called without any password configured for the default user. Are you sure your configuration is correct? [redis.go:2487]
2022/03/28 13:57:40.694892 juicefs[2808338] <INFO>: Data use hdfs://192.168.56.3:9000/pics1/ [format.go:379]
2022/03/28 13:57:52.710700 juicefs[2808338] <FATAL>: Storage hdfs://192.168.56.3:9000/pics1/ is not configured correctly: Failed to put: create /pics1/testing/.v1E5wkkBZh.tmp.6687614270237389643: permission denied [format.go:382]
报错没有权限 谢谢

###### sudo juicefs format --storage hdfs --bucket h1.taosdata.com:9000 --access-key hdfs redis://:123456@127.0.0.1:6379/3 user4
2022/03/28 14:27:57.867419 juicefs[2829940] <INFO>: Meta address: redis://:****@127.0.0.1:6379/3 [interface.go:383]
2022/03/28 14:27:57.868133 juicefs[2829940] <WARNING>: parse info: ERR AUTH <password> called without any password configured for the default user. Are you sure your configuration is correct? [redis.go:2487]
2022/03/28 14:27:57.870046 juicefs[2829940] <INFO>: Data use hdfs://h1.taosdata.com:9000/user4/ [format.go:379]
2022/03/28 14:27:57.901918 juicefs[2829940] <FATAL>: format: ERR AUTH <password> called without any password configured for the default user. Are you sure your configuration is correct? [format.go:403]

######## juicefs mount -d "redis://:123456@127.0.0.1:6379/3" /mnt/hdfs
2022/03/28 14:30:15.875030 juicefs[2832447] <INFO>: Meta address: redis://:****@127.0.0.1:6379/3 [interface.go:383]
2022/03/28 14:30:15.876316 juicefs[2832447] <WARNING>: parse info: ERR AUTH <password> called without any password configured for the default user. Are you sure your configuration is correct? [redis.go:2487]
2022/03/28 14:30:15.876794 juicefs[2832447] <FATAL>: load setting: ERR AUTH <password> called without any password configured for the default user. Are you sure your configuration is correct? [mount.go:172]

**Comment by xiaogaozi** [comment, marked as answer]
> ERR AUTH called without any password configured for the default user. Are you sure your configuration is correct? 

请检查你的 Redis 配置是否正确

**Reply by ssq001** [reply]
https://juicefs.com/docs/zh/community/hadoop_java_sdk#1-%E5%87%BA%E7%8E%B0-class-iojuicefsjuicefilesystem-not-found-%E5%BC%82%E5%B8%B8
谢谢已配置好redis 但是hadoop fs -ls后出现了这个问题 我只有开源的hdfs 应该把jar放在什么位置呢？

**Reply by zhijian-pro** [reply]
放到 $HADOOP_HOME/share/hadoop/common 下面
